base/regex.jl

```diff
@@ -137,7 +137,7 @@ end
 
 call(r::Regex, s) = ismatch(r, s)
 
-function match(re::Regex, str::UTF8String, idx::Integer, add_opts::UInt32=UInt32(0))
+function match(re::Regex, str::Union{SubString{UTF8String}, UTF8String}, idx::Integer, add_opts::UInt32=UInt32(0))
     compile(re)
     opts = re.match_options | add_opts
     if !PCRE.exec(re.regex, str, idx-1, opts, re.match_data)
```

Comment:
Ya, that is the issue. I opted for a somewhat less ad-hoc change that essentially accomplishes that.